src/pluggy/_tracing.py

```diff
@@ -29,24 +28,26 @@ def format_message(self, tags, args):
             lines.append("%s    %s: %s\n" % (indent, name, value))
         return lines
 
-    def processmessage(self, tags, args):
-        if self.writer is not None and args:
-            lines = self.format_message(tags, args)
-            self.writer("".join(lines))
+    def _processmessage(self, tags, args):
+        if self._writer is not None and args:
+            lines = self._format_message(tags, args)
```

Comment:
if this is the only use, lets move the joining into the function to contain the logic

Reply:
I thought to do it too, but `_format_message` has separate tests, so I kept it.

Reply:
@blueyed i'd leave the function and the separate test, but if the string joining is within the format_message function, the rest of the code can be more readable

Reply:
Ha, got it now. I added a commit which does this.